### DISCUSSION TITLE
Setting the PYTHONPATH variable

### DISCUSSION THREAD

**Discussion by gerd35** [pyinstaller/pyinstaller#6031] [Help] [answered]
My frozen app won't run because of a ModuleNotFoundError (encodings). After some research I am quite sure that the fact that there seems to be no PYTHONPATH configured is the reason for this. How do i configure this?

**Comment by rokm** [comment, marked as answer]
The reason is most likely something else. Are you on Apple M1 by any chance?

**Reply by gerd35** [reply]
No, it's an Intel Mac. What else could it be?

**Reply by gerd35** [reply]
This is the output of the executable btw:

Python path configuration:
  PYTHONHOME = '/Users/test/fbs/target/uel.app/Contents/MacOS'
  PYTHONPATH = (not set)
  program name = '/Users/test/fbs/target/uel.app/Contents/MacOS/uel'
  isolated = 0
  environment = 0
  user site = 0
  import site = 0
  sys._base_executable = '/Users/test/fbs/target/uel.app/Contents/MacOS/uel'
  sys.base_prefix = ''
  sys.base_exec_prefix = ''
  sys.platlibdir = 'lib'
  sys.executable = '/Users/test/fbs/target/uel.app/Contents/MacOS/uel'
  sys.prefix = ''
  sys.exec_prefix = ''
  sys.path = [
    '/Users/test/fbs/target/uel.app/Contents/MacOS/base_library.zip',
    '/Users/test/fbs/target/uel.app/Contents/MacOS/lib-dynload',
    '/Users/test/fbs/target/uel.app/Contents/MacOS',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00000001104e9e00 (most recent call first):
<no Python frame>
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

**Reply by rokm** [reply]
Last time I saw missing `encodings` on Mac, it was because we failed to launch python interpreter in helper functions (such as `collect_submodules`), which might have been caused by #6027 (although I would expect that on M1).

When you build your application, do you see any "Bad CPU type" messages? Or any other error messages?

**Reply by rokm** [reply]
Also, can you try running the following commands in your python interpreter (the same one you have pyinstaller installed in):
```python
from PyInstaller.utils.hooks import collect_submodules
collect_submodules('encodings')
```

**Reply by bwoodsend** [reply]
The `encodings` module going missing has always been a red herring disguising something else completely. Are you running the executable in the `build` folder perhaps instead of the `dist` folder? Or moving the executable without the other files? Also how did you install Python?

**Reply by gerd35** [reply]
I use pyinstaller via fbs, hence the mentions of it in the output above. After posting said output here, I tried freezing with pyinstaller directly, which gave a different error output (and the same as in the PyInstaller folder generated by freezing with fbs, which I discovered afterwards):

[64121] Error loading Python lib '/Users/test/fbs/target/PyInstaller/uel/Python': dlopen: dlopen(/Users/test/fbs/target/PyInstaller/uel/Python, 10): image not found
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

**Reply by gerd35** [reply]
Apart from a warning that the hidden import "sip" was not found, there are only INFO and no Error messages when building. I am not running from the build folder or moving the executable and I am pretty sure that I installed python via homebrew. The first command @rokm mentioned succeeds without any output, the second outputs a list of encodings when I run it.

**Reply by rokm** [reply]
> `[64121] Error loading Python lib '/Users/test/fbs/target/PyInstaller/uel/Python': dlopen: dlopen(/Users/test/fbs/target/PyInstaller/uel/Python, 10): image not found`

This is usually the sign of running from the `build` directory instead of `dist` one...

**Reply by rokm** [reply]
What are the contents of `/Users/test/fbs/target/PyInstaller/uel`? And what are the contents of `/Users/test/fbs/target/PyInstaller`?

**Reply by rokm** [reply]
But anyway, I tried to install `fbs` (first time I heard of it) and it force-installed ancient PyInstaller 3.4. Can you check that this is also the case on your sistem? What is the output of `pyinstaller --version`?

**Reply by gerd35** [reply]
/Users/test/fbs/target/PyInstaller/uel:

Analysis-00.toc		PKG-00.pkg		base_library.zip
BUNDLE-00.toc		PKG-00.toc		uel
COLLECT-00.toc		PYZ-00.pyz		warn-uel.txt
EXE-00.toc		PYZ-00.toc		xref-uel.html

/Users/test/fbs/target/PyInstaller:

__pycache__		uel
fbs_pyinstaller_hook.py	uel.spec

Pyinstaller version is indeed 3.4

**Reply by gerd35** [reply]
I guess the logical step to do now is to reinstall pyinstaller and try freezing again right?

**Reply by rokm** [reply]
Yeah, we support only 4.x series. Also, it seems that `fbs` customizes the build process (judging by the `fbs_pyinstaller_hook.py` file and lack of standard `build` and `dist` directories) - so we have no way of supporting that without looking into what they do.

**Comment by gerd35** [comment]
Freezing now works, thank you for the quick replies! It seems as if only the pro Version of fbs supports the newest version of pyinstaller, so I may have to fork the GPL version if I don't find a fork of it or a different program that fulfils my needs.
